clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1269,7 +1269,7 @@ public long position(TopicPartition partition) {
                 throw new IllegalArgumentException("You can only check the position for partitions assigned to this consumer.");
             Long offset = this.subscriptions.position(partition);
             if (offset == null) {
-                updateFetchPositions(Collections.singleton(partition));
+                updateFetchPositions(subscriptions.assignedPartitions());
```

Comment:
What is the motivation for this change? It seems a little odd that we update the fetch positions for all assigned partitions if the position for one partition is `null`.